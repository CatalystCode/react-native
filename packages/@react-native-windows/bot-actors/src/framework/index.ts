/**
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * @format
 */

export * from './ActorEvents';
export * from './ActorInstance';
export * from './ActorRegistry';
export * from './ActorsHandle';
export * from './AsyncEventEmitter';
export * from './Context';
export * from './Mocks';
export * from './initializeActors';
