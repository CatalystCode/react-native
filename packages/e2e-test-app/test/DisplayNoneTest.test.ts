/**
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * @format
 */

import {goToComponentExample, dumpVisualTree} from './framework';

beforeAll(async () => {
  await goToComponentExample('Display:none Style');
});

describe('DisplayNoneTest', () => {
  /* Test case #1: display:none disabled */
  test('DisplayNoneDisabledTest', async () => {
    const dump = await dumpVisualTree('view-component-switch-view');
    expect(dump).toMatchSnapshot();
  });

  /* Test case #2: Enable display:none once, update view*/
  test('DisplayNoneEnabledTest', async () => {
    await toggleDisplayNone();
    const dump = await dumpVisualTree('view-component-switch-view');
    expect(dump).toMatchSnapshot();
  });
});

async function toggleDisplayNone() {
  const showDisplayNoneToggle = await $('~toggle-display:none');
  await showDisplayNoneToggle.click();
}
