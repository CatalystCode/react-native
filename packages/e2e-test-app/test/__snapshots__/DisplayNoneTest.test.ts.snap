// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayNoneTest DisplayNoneDisabledTest 1`] = `
Object {
  "AutomationId": "view-component-switch-view",
  "Background": null,
  "BorderBrush": "#00000000",
  "BorderThickness": "0,0,0,0",
  "Clip": null,
  "CornerRadius": "0,0,0,0",
  "FlowDirection": "LeftToRight",
  "Height": 100,
  "HorizontalAlignment": "Stretch",
  "Left": 0,
  "Margin": "0,0,0,0",
  "Top": 0,
  "VerticalAlignment": "Stretch",
  "Visibility": "Visible",
  "Width": 1074,
  "XamlType": "Microsoft.ReactNative.ViewPanel",
  "children": Array [
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "toggle display:none TextInput1",
      "Top": 0,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 1074,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Background": null,
      "BorderBrush": null,
      "BorderThickness": "0,0,0,0",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 31,
      "HorizontalAlignment": "Left",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Top": 19,
      "VerticalAlignment": "Center",
      "Visibility": "Visible",
      "Width": 51,
      "XamlType": "Windows.UI.Xaml.Controls.ToggleSwitch",
      "children": Array [
        Object {
          "Background": null,
          "BorderBrush": null,
          "BorderThickness": "0,0,0,0",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "HorizontalAlignment": "Stretch",
          "Left": 0,
          "Margin": "0,0,0,0",
          "Padding": "0,0,0,0",
          "Top": 0,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "XamlType": "Windows.UI.Xaml.Controls.Grid",
          "children": Array [
            Object {
              "Background": null,
              "BorderBrush": null,
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "HorizontalAlignment": "Left",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Top",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.Grid",
              "children": Array [
                Object {
                  "Background": "#00FFFFFF",
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,5,0,5",
                  "Name": "SwitchAreaGrid",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Grid",
                },
                Object {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "HorizontalAlignment": "Left",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "OffContentPresenter",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Center",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                },
                Object {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "HorizontalAlignment": "Left",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "OnContentPresenter",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Center",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                },
                Object {
                  "Clip": null,
                  "FlowDirection": "LeftToRight",
                  "Height": 20,
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "OuterBorder",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 40,
                  "XamlType": "Windows.UI.Xaml.Shapes.Rectangle",
                },
                Object {
                  "Clip": null,
                  "FlowDirection": "LeftToRight",
                  "Height": 20,
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "SwitchKnobBounds",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 40,
                  "XamlType": "Windows.UI.Xaml.Shapes.Rectangle",
                },
                Object {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Height": 20,
                  "HorizontalAlignment": "Left",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "SwitchKnob",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 20,
                  "XamlType": "Windows.UI.Xaml.Controls.Grid",
                  "children": Array [
                    Object {
                      "Clip": null,
                      "FlowDirection": "LeftToRight",
                      "Height": 10,
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Name": "SwitchKnobOn",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 10,
                      "XamlType": "Windows.UI.Xaml.Shapes.Ellipse",
                    },
                    Object {
                      "Clip": null,
                      "FlowDirection": "LeftToRight",
                      "Height": 10,
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Name": "SwitchKnobOff",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 10,
                      "XamlType": "Windows.UI.Xaml.Shapes.Ellipse",
                    },
                  ],
                },
                Object {
                  "Background": "#33000000",
                  "BorderBrush": "#00FFFFFF",
                  "BorderThickness": "1,1,1,1",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "SwitchThumb",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Primitives.Thumb",
                  "children": Array [
                    Object {
                      "Clip": null,
                      "FlowDirection": "LeftToRight",
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "XamlType": "Windows.UI.Xaml.Shapes.Rectangle",
                    },
                  ],
                },
              ],
            },
          ],
        },
      ],
    },
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "toggle display:none TextInput2",
      "Top": 50,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 1074,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "AutomationId": "toggle-display:none",
      "Background": null,
      "BorderBrush": null,
      "BorderThickness": "0,0,0,0",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 31,
      "HorizontalAlignment": "Left",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Top": 69,
      "VerticalAlignment": "Center",
      "Visibility": "Visible",
      "Width": 51,
      "XamlType": "Windows.UI.Xaml.Controls.ToggleSwitch",
      "children": Array [
        Object {
          "Background": null,
          "BorderBrush": null,
          "BorderThickness": "0,0,0,0",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "HorizontalAlignment": "Stretch",
          "Left": 0,
          "Margin": "0,0,0,0",
          "Padding": "0,0,0,0",
          "Top": 0,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "XamlType": "Windows.UI.Xaml.Controls.Grid",
          "children": Array [
            Object {
              "Background": null,
              "BorderBrush": null,
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "HorizontalAlignment": "Left",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Top",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.Grid",
              "children": Array [
                Object {
                  "Background": "#00FFFFFF",
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,5,0,5",
                  "Name": "SwitchAreaGrid",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Grid",
                },
                Object {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "HorizontalAlignment": "Left",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "OffContentPresenter",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Center",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                },
                Object {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "HorizontalAlignment": "Left",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "OnContentPresenter",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Center",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                },
                Object {
                  "Clip": null,
                  "FlowDirection": "LeftToRight",
                  "Height": 20,
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "OuterBorder",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 40,
                  "XamlType": "Windows.UI.Xaml.Shapes.Rectangle",
                },
                Object {
                  "Clip": null,
                  "FlowDirection": "LeftToRight",
                  "Height": 20,
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "SwitchKnobBounds",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 40,
                  "XamlType": "Windows.UI.Xaml.Shapes.Rectangle",
                },
                Object {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Height": 20,
                  "HorizontalAlignment": "Left",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "SwitchKnob",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 20,
                  "XamlType": "Windows.UI.Xaml.Controls.Grid",
                  "children": Array [
                    Object {
                      "Clip": null,
                      "FlowDirection": "LeftToRight",
                      "Height": 10,
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Name": "SwitchKnobOn",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 10,
                      "XamlType": "Windows.UI.Xaml.Shapes.Ellipse",
                    },
                    Object {
                      "Clip": null,
                      "FlowDirection": "LeftToRight",
                      "Height": 10,
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Name": "SwitchKnobOff",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 10,
                      "XamlType": "Windows.UI.Xaml.Shapes.Ellipse",
                    },
                  ],
                },
                Object {
                  "Background": "#33000000",
                  "BorderBrush": "#00FFFFFF",
                  "BorderThickness": "1,1,1,1",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "SwitchThumb",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Primitives.Thumb",
                  "children": Array [
                    Object {
                      "Clip": null,
                      "FlowDirection": "LeftToRight",
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "XamlType": "Windows.UI.Xaml.Shapes.Rectangle",
                    },
                  ],
                },
              ],
            },
          ],
        },
      ],
    },
  ],
}
`;

exports[`DisplayNoneTest DisplayNoneEnabledTest 1`] = `
Object {
  "AutomationId": "view-component-switch-view",
  "Background": null,
  "BorderBrush": "#00000000",
  "BorderThickness": "0,0,0,0",
  "Clip": null,
  "CornerRadius": "0,0,0,0",
  "FlowDirection": "LeftToRight",
  "Height": 100,
  "HorizontalAlignment": "Stretch",
  "Left": 0,
  "Margin": "0,0,0,0",
  "Top": 0,
  "VerticalAlignment": "Stretch",
  "Visibility": "Visible",
  "Width": 1074,
  "XamlType": "Microsoft.ReactNative.ViewPanel",
  "children": Array [
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "toggle display:none TextInput1",
      "Top": 0,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 1074,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "Background": null,
      "BorderBrush": null,
      "BorderThickness": "0,0,0,0",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 31,
      "HorizontalAlignment": "Left",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Top": 19,
      "VerticalAlignment": "Center",
      "Visibility": "Visible",
      "Width": 51,
      "XamlType": "Windows.UI.Xaml.Controls.ToggleSwitch",
      "children": Array [
        Object {
          "Background": null,
          "BorderBrush": null,
          "BorderThickness": "0,0,0,0",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "HorizontalAlignment": "Stretch",
          "Left": 0,
          "Margin": "0,0,0,0",
          "Padding": "0,0,0,0",
          "Top": 0,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "XamlType": "Windows.UI.Xaml.Controls.Grid",
          "children": Array [
            Object {
              "Background": null,
              "BorderBrush": null,
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "HorizontalAlignment": "Left",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Top",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.Grid",
              "children": Array [
                Object {
                  "Background": "#00FFFFFF",
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,5,0,5",
                  "Name": "SwitchAreaGrid",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Grid",
                },
                Object {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "HorizontalAlignment": "Left",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "OffContentPresenter",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Center",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                },
                Object {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "HorizontalAlignment": "Left",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "OnContentPresenter",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Center",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                },
                Object {
                  "Clip": null,
                  "FlowDirection": "LeftToRight",
                  "Height": 20,
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "OuterBorder",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 40,
                  "XamlType": "Windows.UI.Xaml.Shapes.Rectangle",
                },
                Object {
                  "Clip": null,
                  "FlowDirection": "LeftToRight",
                  "Height": 20,
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "SwitchKnobBounds",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 40,
                  "XamlType": "Windows.UI.Xaml.Shapes.Rectangle",
                },
                Object {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Height": 20,
                  "HorizontalAlignment": "Left",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "SwitchKnob",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 20,
                  "XamlType": "Windows.UI.Xaml.Controls.Grid",
                  "children": Array [
                    Object {
                      "Clip": null,
                      "FlowDirection": "LeftToRight",
                      "Height": 10,
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Name": "SwitchKnobOn",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 10,
                      "XamlType": "Windows.UI.Xaml.Shapes.Ellipse",
                    },
                    Object {
                      "Clip": null,
                      "FlowDirection": "LeftToRight",
                      "Height": 10,
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Name": "SwitchKnobOff",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 10,
                      "XamlType": "Windows.UI.Xaml.Shapes.Ellipse",
                    },
                  ],
                },
                Object {
                  "Background": "#33000000",
                  "BorderBrush": "#00FFFFFF",
                  "BorderThickness": "1,1,1,1",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "SwitchThumb",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Primitives.Thumb",
                  "children": Array [
                    Object {
                      "Clip": null,
                      "FlowDirection": "LeftToRight",
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "XamlType": "Windows.UI.Xaml.Shapes.Rectangle",
                    },
                  ],
                },
              ],
            },
          ],
        },
      ],
    },
    Object {
      "Clip": null,
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 19,
      "HorizontalAlignment": "Stretch",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Text": "toggle display:none TextInput2",
      "Top": 50,
      "VerticalAlignment": "Stretch",
      "Visibility": "Visible",
      "Width": 1074,
      "XamlType": "Windows.UI.Xaml.Controls.TextBlock",
    },
    Object {
      "AutomationId": "toggle-display:none",
      "Background": null,
      "BorderBrush": null,
      "BorderThickness": "0,0,0,0",
      "Clip": null,
      "CornerRadius": "0,0,0,0",
      "FlowDirection": "LeftToRight",
      "Foreground": "#FF000000",
      "Height": 31,
      "HorizontalAlignment": "Left",
      "Left": 0,
      "Margin": "0,0,0,0",
      "Padding": "0,0,0,0",
      "Top": 69,
      "VerticalAlignment": "Center",
      "Visibility": "Visible",
      "Width": 51,
      "XamlType": "Windows.UI.Xaml.Controls.ToggleSwitch",
      "children": Array [
        Object {
          "Background": null,
          "BorderBrush": null,
          "BorderThickness": "0,0,0,0",
          "Clip": null,
          "CornerRadius": "0,0,0,0",
          "FlowDirection": "LeftToRight",
          "HorizontalAlignment": "Stretch",
          "Left": 0,
          "Margin": "0,0,0,0",
          "Padding": "0,0,0,0",
          "Top": 0,
          "VerticalAlignment": "Stretch",
          "Visibility": "Visible",
          "XamlType": "Windows.UI.Xaml.Controls.Grid",
          "children": Array [
            Object {
              "Background": null,
              "BorderBrush": null,
              "BorderThickness": "0,0,0,0",
              "Clip": null,
              "CornerRadius": "0,0,0,0",
              "FlowDirection": "LeftToRight",
              "HorizontalAlignment": "Left",
              "Left": 0,
              "Margin": "0,0,0,0",
              "Padding": "0,0,0,0",
              "Top": 0,
              "VerticalAlignment": "Top",
              "Visibility": "Visible",
              "XamlType": "Windows.UI.Xaml.Controls.Grid",
              "children": Array [
                Object {
                  "Background": "#00FFFFFF",
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,5,0,5",
                  "Name": "SwitchAreaGrid",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Grid",
                },
                Object {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "HorizontalAlignment": "Left",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "OffContentPresenter",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Center",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                },
                Object {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "HorizontalAlignment": "Left",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "OnContentPresenter",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Center",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.ContentPresenter",
                },
                Object {
                  "Clip": null,
                  "FlowDirection": "LeftToRight",
                  "Height": 20,
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "OuterBorder",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 40,
                  "XamlType": "Windows.UI.Xaml.Shapes.Rectangle",
                },
                Object {
                  "Clip": null,
                  "FlowDirection": "LeftToRight",
                  "Height": 20,
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "SwitchKnobBounds",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 40,
                  "XamlType": "Windows.UI.Xaml.Shapes.Rectangle",
                },
                Object {
                  "Background": null,
                  "BorderBrush": null,
                  "BorderThickness": "0,0,0,0",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Height": 20,
                  "HorizontalAlignment": "Left",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "SwitchKnob",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "Width": 20,
                  "XamlType": "Windows.UI.Xaml.Controls.Grid",
                  "children": Array [
                    Object {
                      "Clip": null,
                      "FlowDirection": "LeftToRight",
                      "Height": 10,
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Name": "SwitchKnobOn",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 10,
                      "XamlType": "Windows.UI.Xaml.Shapes.Ellipse",
                    },
                    Object {
                      "Clip": null,
                      "FlowDirection": "LeftToRight",
                      "Height": 10,
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Name": "SwitchKnobOff",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "Width": 10,
                      "XamlType": "Windows.UI.Xaml.Shapes.Ellipse",
                    },
                  ],
                },
                Object {
                  "Background": "#33000000",
                  "BorderBrush": "#00FFFFFF",
                  "BorderThickness": "1,1,1,1",
                  "Clip": null,
                  "CornerRadius": "0,0,0,0",
                  "FlowDirection": "LeftToRight",
                  "Foreground": "#FF000000",
                  "HorizontalAlignment": "Stretch",
                  "Left": 0,
                  "Margin": "0,0,0,0",
                  "Name": "SwitchThumb",
                  "Padding": "0,0,0,0",
                  "Top": 0,
                  "VerticalAlignment": "Stretch",
                  "Visibility": "Visible",
                  "XamlType": "Windows.UI.Xaml.Controls.Primitives.Thumb",
                  "children": Array [
                    Object {
                      "Clip": null,
                      "FlowDirection": "LeftToRight",
                      "HorizontalAlignment": "Stretch",
                      "Left": 0,
                      "Margin": "0,0,0,0",
                      "Top": 0,
                      "VerticalAlignment": "Stretch",
                      "Visibility": "Visible",
                      "XamlType": "Windows.UI.Xaml.Shapes.Rectangle",
                    },
                  ],
                },
              ],
            },
          ],
        },
      ],
    },
  ],
}
`;
